### PR TITLE
Packages+tsconfig

### PR DESCRIPTION
**Pull request by jolene-loz**
Packages and tsconfig json files. Also added the gts and tslint packages as suggested

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "prepare": "npm run compile",
+    "pretest": "npm run compile",
+    "posttest": "npm run check"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/googleinterns/smart-home-testing.git"
+  },
+  "author": "Jolene Lozano",
+  "license": "ISC",
```
Comment (Fleker) [inline package.json:21]: The license is Apache-2.0

File: package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  "bugs": {
+    "url": "https://github.com/googleinterns/smart-home-testing/issues"
+  },
+  "homepage": "https://github.com/googleinterns/smart-home-testing#readme",
+  "devDependencies": {
+    "chai-json-schema-ajv": "^5.1.0",
+    "mocha": "^8.0.1",
+    "mocha-param": "^2.0.1",
+    "ts-node": "^8.10.2",
```
Comment (devunwired) [inline package.json:30]: Is this used? Planning to use?

File: tslint.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (devunwired) [inline tslint.json:1]: If you're using `gts` for lint/style, I believe that's based on eslint and doesn't use tslint at all?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    "fix": "gts fix",
+    "prepare": "npm run compile",
+    "pretest": "npm run compile",
+    "posttest": "npm run check"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/googleinterns/smart-home-testing.git"
+  },
+  "author": "Jolene Lozano",
```
Comment (devunwired) [inline package.json:20]: @Fleker Not sure on project rules, should this be Google or the team name or the individual author?

File: package.json (hunk 2 above)
Comment (jolene-loz) [reply package.json:30]: This is here for the example developer test - could remove this though for this PR

**Comment by jolene-loz** [comment]
> This PR is a bit confusing since it includes the commits from #1 as well. Is it possible to isolate this to just the relevant changes?

Yes! Should've caught this before

File: package.json (hunk 4 above)
Comment (Fleker) [reply package.json:20]: Formally it should be "Google LLC".

File: package.json (hunk 2 above)
Comment (Fleker) [reply package.json:30]: Yeah we should make each commit atomic, only including what is required. Later dependencies can be included in the future alongside an example developer test.

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "./node_modules/gts/tsconfig-google.json",
```
Comment (Fleker) [inline tsconfig.json:2]: Does this miss dependencies in the `package.json` for importing the linting libraries/tools?

File: src/validate.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+// import {v4 as uuidv4} from 'uuid';
```
Comment (Fleker) [inline src/validate.ts:1]: `git rm --cached src/validate.ts`

File: README.md [hunk 7]
```diff
@@ -3,3 +3,10 @@
 ## Disclaimer
 
 **This is not an officially supported Google product.**
+
+##Testing Library for Smart Home
```
Comment (Fleker) [inline README.md:7]: Typically a space is placed after the pound signs

File: tsconfig.json (hunk 5 above)
Comment (jolene-loz) [reply tsconfig.json:3]: Could you elaborate on this question a little more? What dependencies are you referring to?

Comment (Fleker) [reply tsconfig.json:2]: Nevermind I think this is okay now

File: README.md [hunk 8]
```diff
@@ -1,5 +0,0 @@
-# smarthome-testing
```
Comment (Fleker) [inline README.md:1]: Can we undo the deletion of this file?

File: package.json [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/googleinterns/smart-home-testing.git"
+  },
+  "author": "Google LLC",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "https://github.com/googleinterns/smart-home-testing/issues"
+  },
+<<<<<<< HEAD
```
Comment (Fleker) [inline package.json:28]: It looks like there are merge conflicts again

Comment (Fleker) [reply package.json:28]: There still seem to be merge conflicts, as these characters should not exist in a valid JSON file.

File: package.json [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+>>>>>>> Redoing commits
+=======
+  "homepage": "https://github.com/googleinterns/smart-home-testing#readme",
+  "devDependencies": {
+    "gts": "^2.0.2",
+    "typescript": "^3.8.3",
+    "@types/node": "^13.11.1"
+  }
+}
+>>>>>>> Edits to package + tsconfig
```
[the remark this answers is not in the document]
Comment (jolene-loz) [reply package.json:41]: I don't get these errors in my IDE, is there another place that I can find these errors? It shows that the merge conflicts are resolved on my end

[the remark this answers is not in the document]
Comment (Fleker) [reply package.json:41]: The conflicts have been merged to an extent, although there are these text artifacts that remain.

https://jsonlint.com is one tool that can verify that file is valid.

**Pull request closed without merging**